circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -531,6 +558,68 @@ private enum class InstantiationType {
   CLASS
 }
 
+private enum class CodegenMode {
+  ANVIL {
+    override fun annotateFactory(builder: TypeSpec.Builder, scope: TypeName) {
+      builder.addAnnotation(
+        AnnotationSpec.builder(ContributesMultibinding::class).addMember("%T::class", scope).build()
+      )
+    }
+  },
+  HILT {
+    override fun produceAdditionalTypeSpec(
+      factory: ClassName,
+      factoryType: FactoryType,
+      scope: TypeName
+    ): TypeSpec {
+      val moduleAnnotations =
```

Comment:
Just realised there may be a roadblock in all of this. Here we're missing an `@OriginatingElement`, which should be trivial to add. 

The more troubling bit is `@GeneratesRootInput`, which is needed as meta annotation on `@CircuitInject` for Hilt to delay component generation properly.  Given circuit is a multiplatform project, the only way I can think to sort this out is through an `androidMain` source-set for the annotation module. A broader change, and more maintenance to keep up with as Dagger tends towards supporting ksp & multiplatform. Thoughts?

Reply:
Limiting it to just androidMain is fine with me. `SymbolProcessorEnvironment` has a `platforms` property that we can check to enforce this, just would need to add a corresponding set to compare to in the enum. The anvil impl would only support android and jvm as well.